ui/src/heap/AddCurioModal.tsx

```diff
@@ -45,39 +48,42 @@ export default function AddCurioModal({
   const { privacy } = useGroupPrivacy(flag);
 
   const isEmpty = !content && !draggedFile && !pastedFile;
+  const errorForDisplay = errorMessage || dragErrorMessage;
 
   const reset = useCallback(() => {
     setMode('input');
     setContent(undefined);
     setPreviewUrl('');
     setStatus('initial');
-    if (draggedFile) {
-      clearDraggedFile();
-    }
-    if (pastedFile) {
-      setPastedFile(null);
-    }
-  }, [clearDraggedFile, draggedFile, pastedFile]);
+    clearDragState();
+    setPastedFile(null);
+    setErrorMessage('');
+  }, [clearDragState]);
```

Comment:
Just calling out not sure if worth doing now though, probably could combine these into a richer object to make these state transitions cleaner.

Reply:
Fair, state balooned a bit. I don't think this should block merge, but I'll combine it when I have a chance.